res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -1444,69 +1444,132 @@ let wheelTimerDelta = 0;
+        for (const num of numbers) {
+            if (typeof num === "number") {
+                numCount++;
+            }
+        }
+        if (numCount < numbers.length) {
+            console.warn("ERROR! new S4Mk3EffectUnit() called with array that contains other types than numbers!");
+            return;
+        }
+        this.unitNumbers = numbers;
+
+        this.unitNumber = this.unitNumbers[0];
+        this.group = this.unitGroupForNumber(this.unitNumber);
         this.focusedEffect = null;
+        this.focusSelectMode = false;
+        // With this control skins can mark the unit that is currently mapped to the controller
+        engine.setValue(this.group, "controller_input_active", 1);
+        // Unset for the other unit
+        const otherUnitGroup = this.unitGroupForNumber(this.unitNumbers[1]);
+        engine.setValue(otherUnitGroup, "controller_input_active", 0);
+        // From now on we only toggle it in setCurrentUnit()
 
         this.mixKnob = new Pot({
             inKey: "mix",
             group: this.group,
             inReport: inReports[2],
             inByte: io.mixKnob.inByte,
+            // TODO Super knob when shifted?
         });
 
-        this.mainButton = new PowerWindowButton({
+        // Focus button
+        // shortpress: clear focused effect, return to Meta/toggle mode
+        // shortpress + Shift: switch unit
+        // longpress: focus select mode
+        // LED: blinking in focus mode, else off
+        this.effectFocusButton = new PowerWindowButton({
             unit: this,
+            group: this.group,
             inReport: inReports[1],
-            inByte: io.mainButton.inByte,
-            inBit: io.mainButton.inBit,
-            outByte: io.mainButton.outByte,
+            inByte: io.effectFocusButton.inByte,
+            inBit: io.effectFocusButton.inBit,
+            outByte: io.effectFocusButton.outByte,
             outReport: outReport,
-            shift: function() {
-                this.group = this.unit.group;
-                this.outKey = "group_[Master]_enable";
-                this.outConnect();
-                this.outTrigger();
-            },
-            unshift: function() {
-                this.outDisconnect();
-                this.outKey = undefined;
-                this.group = undefined;
-                this.output(false);
-            },
-            input: function(pressed) {
-                if (!this.shifted) {
-                    for (const index of [0, 1, 2]) {
-                        const effectGroup = `[EffectRack1_EffectUnit${unitNumber}_Effect${index + 1}]`;
-                        engine.setValue(effectGroup, "enabled", pressed);
+            // onShortPress: function() {
+            //     // TODO dummy impl to avoid "Unknown control [EffectRack1_EffectUnit1], "
+            //     // TODO seems this slips through in Button -> defaultInput()
+            //     // console.warn(`focusButton shortPress`);
+            // },
```

Comment:
Is this needed?

Reply:
Well, I saw that warning being logged.
But I'll try again.